features/templates.feature

```diff
@@ -1,7 +1,42 @@
 @templates
 Feature: Templates
 
-    Description from screwdriver-cd/screwdriver#470
+    Templates are simply an existing configuration for a job that can be reused.
+    A basic example would be something like this:
+        name: node/test
+        version: 1.0.0
+        description: Template for testing
+        maintainer: foo@bar.com
+        config:
+            image: node:6
+            steps:
+                - install: npm install
+                - test: npm test
+
+    To use a template, users would consume it like this in their screwdriver.yaml:
+        jobs:
+            main:
+                template: node/test@1
+
+    The user's config would be translated to:
+        jobs:
+            main:
+                image: node:6
+                steps:
+                    - install: npm install
+                    - test: npm test
+
+    Rules:
+        - Template owner can publish a template if
+              - The format of template is valid
+              - The build that is used to publish has the right permissions
+              - No existing version already exists
```

Comment:
We will auto bump the version and ignore the patch version, so this will not happen